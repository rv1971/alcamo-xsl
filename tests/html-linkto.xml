<?xml version="1.0" encoding="UTF-8"?>

<?xml-stylesheet href="#xslt" type="text/xml"?>

<a:doc
    xmlns:dc="http://purl.org/dc/terms/"
    xmlns:owl="http://www.w3.org/2002/07/owl#"
    xmlns:a="tag:rv1971@web.de,2021:alcamo-xsl#">
  <xsl:stylesheet
      xmlns="http://www.w3.org/1999/xhtml"
      xmlns:dc="http://purl.org/dc/terms/"
      xmlns:owl="http://www.w3.org/2002/07/owl#"
      xmlns:xsl="http://www.w3.org/1999/XSL/Transform"
      xml:id="xslt"
      version="1.0"
      exclude-result-prefixes="dc owl">
    <xsl:import href="../xsl/html.xsl"/>

    <xsl:template match="/">
      <a:result>
        <xsl:apply-templates select="/a:doc/a:data"/>
     </a:result>
    </xsl:template>

    <xsl:template match="a:item">
      <a:item>
        <xsl:value-of select="@*"/>
        <xsl:value-of select="' '"/>
        <xsl:apply-templates select="@*" mode="a:auto"/>
      </a:item>
    </xsl:template>

    <xsl:template match="dc:*|owl:*">
      <a:item>
        <xsl:value-of select="."/>
        <xsl:value-of select="' '"/>
        <xsl:apply-templates select="." mode="a:auto"/>
      </a:item>
    </xsl:template>
  </xsl:stylesheet>

  <a:data>
    <dc:rights>http://example.com/rights</dc:rights>
    <dc:rights>© Alice</dc:rights>
    <owl:sameAs>tag:bob@example.com,2023:foo</owl:sameAs>
    <a:item dc:rights="https://example.com/rights"/>
    <a:item dc:rights="© Bob 1971"/>
    <a:item owl:sameAs="https://example.com/testcase1"/>
  </a:data>
</a:doc>
