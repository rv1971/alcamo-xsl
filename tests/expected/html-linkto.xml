<?xml version="1.0"?>
<a:result xmlns:a="tag:rv1971@web.de,2021:alcamo-xsl#" xmlns="http://www.w3.org/1999/xhtml">
  <a:item>http://example.com/rights <a href="http://example.com/rights">example.com/rights</a></a:item>
  <a:item>&#xA9; Alice &#xA9; Alice</a:item>
  <a:item>tag:bob@example.com,2023:foo tag:bob@example.com,2023:foo</a:item>
  <a:item>https://example.com/rights <a href="https://example.com/rights">example.com/rights</a></a:item>
  <a:item>&#xA9; Bob 1971 &#xA9; Bob 1971</a:item>
  <a:item>https://example.com/testcase1 <a href="https://example.com/testcase1">example.com/testcase1</a></a:item>
</a:result>
